### PR TITLE
refactor: priority of language set by generate_otp

### PR DESCRIPTION
**Pull request by e-lobo**
Priority of language set (frappe.local.lang) in [generate_otp](https://github.com/leam-tech/renovation_core/blob/e66ec1c3863ccffaa65d2582e7eadc06f837c54b/renovation_core/utils/auth.py#L17):
  1. `lang` kwarg
  2. `Accept-Language` header or from `frappe.session.user` (done by **frappe** )
  3. is `user` exists from `medium_id` (_email_ or _mobile_) ie `user.lang` (ie **Guest Users**)

### CODE REVIEW

File: renovation_core/utils/auth.py [hunk 1]
```diff
@@ -37,10 +42,16 @@ def generate_otp(medium="sms", medium_id=None, sms_hash=None, purpose="login", l
...
+  if not frappe.get_request_header("Accept-Language"):
+    # check if existing user and look for lang
+    user = get_linked_user(id_type=medium, id=medium_id)
+    user_lang = None
+    if user:
+      user_lang = frappe.db.get_value("User", user, "language")
+    if frappe.db.exists("Language", lang):
+      frappe.local.lang = lang
+    elif user_lang:
+      frappe.local.lang = user_lang
```
Comment (fahimalizain) [inline renovation_core/utils/auth.py:54]: @e-lobo Let's take this opportunity to revert back to original `frappe.local.lang` after the OTP has been sent 😄

File: renovation_core/utils/auth.py [hunk 2]
```diff
@@ -29,6 +29,11 @@ def generate_otp(medium="sms", medium_id=None, sms_hash=None, purpose="login", l
   :param sms_hash: The hash that should be appended to OTP SMS
   :param purpose: Specify an optional purpose (login, pwd_reset) to make a custom context
   :param lang: Language of the OTP message (SMS or Email)
+
+  Priority of language set (frappe.local.lang):
+  1. Accept-Language header
```
Comment (fahimalizain) [inline renovation_core/utils/auth.py:34]: @e-lobo I think the order should be:
- lang kwarg
- Accept-Lang header
- User obtained from medium_id

File: renovation_core/utils/auth.py (hunk 1 above)
Comment (e-lobo) [reply renovation_core/utils/auth.py]: @fahimalizain done 👍

File: renovation_core/utils/auth.py (hunk 2 above)
Comment (e-lobo) [reply renovation_core/utils/auth.py]: @fahimalizain refactored 👍

**Review by fahimalizain — APPROVED** [review]
👍🏼 👍🏼 👍🏼

**Review by Abadulrehman — APPROVED** [review]
:100:

File: renovation_core/utils/auth.py [hunk 3]
```diff
@@ -29,18 +29,34 @@ def generate_otp(medium="sms", medium_id=None, sms_hash=None, purpose="login", l
   :param sms_hash: The hash that should be appended to OTP SMS
   :param purpose: Specify an optional purpose (login, pwd_reset) to make a custom context
   :param lang: Language of the OTP message (SMS or Email)
+
+  Priority of language set (frappe.local.lang):
+  1. lang kwarg
+  2. Accept-Language header or from frappe.session.user (done by frappe )
+  3. is user exists from medium_id (email or mobile) ie user.lang (ie Guest Users)
```
Comment (assemmarwan) [inline renovation_core/utils/auth.py:36]: @e-lobo Did you mean "not Guest Users"?

Comment (e-lobo) [reply renovation_core/utils/auth.py:36]: @assemmarwan nope.. Guest users will only hit 3

**Review by assemmarwan — APPROVED** [review]

**Pull request merged**